small_vector.h: warning C5232: in C++20 this comparison calls ... recursively

In CMake project `myproject`, fetching and building dependency Dawn v20251215.205450 via CPM, with option `DAWN_FETCH_DEPENDENCIES ON`, which in turn fetches and builds spirv-tools.
Build fails in spirv-tools in `source/util/small_vector.h` with `warning C5232: in C++20 this comparison calls 'bool spvtools::utils::operator ==(const std::vector<uint32_t,std::allocator<uint32_t>> &,const spvtools::utils::SmallVector<uint32_t,2> &)' recursively`.

```
-- Generating done (3.0s)
-- Build files have been written to: C:/myproject/out/build/windows-msvc-debug-developer-mode

[18/1340] Building CXX object _deps\dawn-build\third_party\spirv-tools\source\opt\CMakeFiles\SPIRV-Tools-opt.dir\set_spec_constant_default_value_pass.cpp.obj
FAILED: [code=2] _deps/dawn-build/third_party/spirv-tools/source/opt/CMakeFiles/SPIRV-Tools-opt.dir/set_spec_constant_default_value_pass.cpp.obj
C:\PROGRA~2\MICROS~2\18\BUILDT~1\VC\Tools\MSVC\1450~1.357\bin\Hostx64\x64\cl.exe  /nologo /TP -DSPIRV_COLOR_TERMINAL -DSPIRV_WINDOWS -IC:\myproject\out\build\windows-msvc-debug-developer-mode\_deps\dawn-src\third_party\spirv-tools\src -IC:\myproject\out\build\windows-msvc-debug-developer-mode\_deps\dawn-src\third_party\spirv-tools\src\include -IC:\myproject\out\build\windows-msvc-debug-developer-mode\_deps\dawn-src\third_party\spirv-headers\src\include -IC:\myproject\out\build\windows-msvc-debug-developer-mode\_deps\dawn-build\third_party\spirv-tools /DWIN32 /D_WINDOWS /EHsc  /sdl /DYNAMICBASE /guard:cf  /MP /Ob0 /Od -std:c++20 -MDd -RTC1 -Zi /GL   /MP4 -DSPIRV_CHECK_CONTEXT -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS /wd4800 /wd4819 /wd4251 /W2 /WX /EHs /showIncludes /Fo_deps\dawn-build\third_party\spirv-tools\source\opt\CMakeFiles\SPIRV-Tools-opt.dir\set_spec_constant_default_value_pass.cpp.obj /Fd_deps\dawn-build\third_party\spirv-tools\source\opt\CMakeFiles\SPIRV-Tools-opt.dir\ /FS -c C:\myproject\out\build\windows-msvc-debug-developer-mode\_deps\dawn-src\third_party\spirv-tools\src\source\opt\set_spec_constant_default_value_pass.cpp
C:\myproject\out\build\windows-msvc-debug-developer-mode\_deps\dawn-src\third_party\spirv-tools\src\source/util/small_vector.h(189): error C2220: the following warning is treated as an error
C:\myproject\out\build\windows-msvc-debug-developer-mode\_deps\dawn-src\third_party\spirv-tools\src\source/util/small_vector.h(189): warning C5232: in C++20 this comparison calls 'bool spvtools::utils::operator ==(const std::vector<uint32_t,std::allocator<uint32_t>> &,const spvtools::utils::SmallVector<uint32_t,2> &)' recursively
C:\myproject\out\build\windows-msvc-debug-developer-mode\_deps\dawn-src\third_party\spirv-tools\src\source/util/small_vector.h(189): note: to simplify migration, consider the temporary use of /Wv:18 flag with the version of the compiler with which you used to build without warnings
C:\myproject\out\build\windows-msvc-debug-developer-mode\_deps\dawn-src\third_party\spirv-tools\src\source/util/small_vector.h(189): note: the template instantiation context (the oldest one first) is
C:\myproject\out\build\windows-msvc-debug-developer-mode\_deps\dawn-src\third_party\spirv-tools\src\source/opt/instruction.h(93): note: see reference to class template instantiation 'spvtools::utils::SmallVector<uint32_t,2>' being compiled
C:\myproject\out\build\windows-msvc-debug-developer-mode\_deps\dawn-src\third_party\spirv-tools\src\source/util/small_vector.h(188): note: while compiling friend function 'bool spvtools::utils::operator ==(const std::vector<uint32_t,std::allocator<uint32_t>> &,const spvtools::utils::SmallVector<uint32_t,2> &)'
C:\myproject\out\build\windows-msvc-debug-developer-mode\_deps\dawn-src\third_party\spirv-tools\src\source/util/small_vector.h(194): note: see the first reference to 'spvtools::utils::operator ==' in 'spvtools::utils::operator !='
C:\myproject\out\build\windows-msvc-debug-developer-mode\_deps\dawn-src\third_party\spirv-tools\src\source\opt\set_spec_constant_default_value_pass.cpp(304): note: see the first reference to 'spvtools::utils::operator !=' in 'spvtools::opt::SetSpecConstantDefaultValuePass::Process'
[31/1340] Building CXX object _deps\dawn-build\src\tint\CMakeFiles\tint_lang_spirv_writer_raise.dir\lang\spirv\writer\raise\raise.cc.obj
ninja: build stopped: subcommand failed.
error: Recipe `build` failed on line 10 with exit code 2
```

--- a/source/util/small_vector.h
+++ b/source/util/small_vector.h
@@ -184,7 +184,8 @@ class SmallVector {
   }
 
 // Avoid infinite recursion from rewritten operators in C++20
-#if __cplusplus <= 201703L
+#if (defined(_MSVC_LANG) && _MSVC_LANG <= 201703L) || \
+    (!defined(_MSVC_LANG) && __cplusplus <= 201703L)
   friend bool operator==(const std::vector<T>& lhs, const SmallVector& rhs) {
     return rhs == lhs;
   }

